scripts/abi_artifact_generator/templates/setup_return_values.inc.jinja_template

```diff
@@ -0,0 +1,35 @@
+/*#-
+This template file is distributed under the MIT License. See LICENSE.md for details.
+#*/
+
+/*(- generation_notice )*/
+
+#include "../functions.h"
+
+/* NOTE: An initial value is provided to prevent these from ending up in `.bss`.
+ *       It should never be used.
+ */
+uint8_t randomized_state[/*( generated_byte_count )*/] = { 1 };
+uint8_t printable_location[/*( generated_byte_count )*/] = { 1 };
+
+// A place to safely store tested function return addresses, so they can be
+// recovered no matter how much stack and registers are modified.
+void *saved_return_address;
+
+/*{ for name, return_value_type in return_value_functions.items() -}*/
+void setup_/*( name )*/(void) {
+  typedef /*( return_value_type )*/ return_value_type;
+
+  // Call the tested function.
+  return_value_type returned = test_/*( name )*/();
+
+  // Write the pointer to the received values into a safe location,
+  *(return_value_type **) printable_location = &returned;
+
+  // Then write the value to it as well.
+  *(return_value_type *) (printable_location + 8) = returned;
```

Comment:
I prefer not to rely onto `libc` if we can avoid it, which we currently have.

Also, this way it's more natural - since we leave it to the compiler to copy the object "naturally", which in my experience sometimes leaves out the padding (which is a problem, but I'd say not a big one).

Reply:
I'm not pushing for `memcpy`, I'm asking if the C standards allows to copy `struct`s. But apparently yes, so, we're happy!

Reply:
This! This exact thing makes me so happy I'm a c++ programmer. It never even crossed my mind that there could even exist a world where you giving compiler exact type information is not enough for it to figure out a way to copy an object the right way (tm).